sdjournal/journal.go

```diff
@@ -107,6 +107,39 @@ func (j *Journal) AddMatch(match string) error {
 	return nil
 }
 
+// AddDisjunction inserts a logical OR in the match list.
+func (j *Journal) AddDisjunction() (int, error) {
```

Comment:
The return value for this and AddConjunction should just be `error` - it's redundant to have `int` as well when it's only used to propagate the return code. (I assume you were emulating the signature of `Next`, but I would argue that its current behaviour in the case of failure -- of returning both a non-zero int and a non-nil error -- is unidiomatic, since Go doesn't typically overload errors into return values the same way as in C).

Unfortunately, that brings up the fact that `AddMatch` isn't properly checking return codes right now and only ever returns `nil`. Would you mind fixing that too? Thanks!